File: frontEnd/Employee_System/src/app/modules/employee/services/employee.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseUrl ="http://localhost:8080/api/employee"


  constructor(private httpClient : HttpClient) {}

  getEmployeeById(id:number):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/employees/${id}`, {
      headers: this.createHeaders()
    });
  }

  updateEmployee(id:number , employee :any):Observable<any>{
    return this.httpClient.put(`${this.baseUrl}/employees/${id}`, employee, {
      headers: this.createHeaders()
    });
  }

  ///////////////////////  Tasks ////////////////////////////////

  getTasksByEmpId():Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/tasks`, {
      headers: this.createHeaders()
    });
  }

  getTaskById(id:number):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/task/${id}` ,{
      headers: this.createHeaders()
    })
  }

  upateTaskStatus(id:number , status :string):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/task/${id}/${status}`, {
      headers: this.createHeaders()
    });
  }

  updateTask(id:number , taskDto :any):Observable<any>{
    return this.httpClient.put(`${this.baseUrl}/task/${id}` , taskDto ,{
      headers:this.createHeaders()
    })
  }


  //////////////////////  Comments ///////////////////////

createComment(taskId:number , content:string):Observable<any>{

  const params ={
    content:content
  }
  return this.httpClient.post(`${this.baseUrl}/task/comment/${taskId}` , null,{
    params: params,
    headers:this.createHeaders()
  })
}

getAllCommentsOfTask(id:number):Observable<any>{
  return this.httpClient.get(`${this.baseUrl}/comments/${id}` ,{
    headers: this.createHeaders()
  })
}



  private createHeaders():HttpHeaders{
    return new HttpHeaders().set(
      "Authorization" , "Bearer "+StorageService.getToken());
    }



    
}

File: employee-backend/src/main/java/com/employee/service/EmployeeService.java
package com.employee.service;

import java.util.List;

import org.springframework.security.core.userdetails.UserDetailsService;

import com.employee.Dto.CommentDto;
import com.employee.Dto.EmployeeDto;
import com.employee.Dto.TaskDto;
import com.employee.model.Employee;

public interface EmployeeService {
	
	UserDetailsService userDetailsService();
	
	 
	 
	 Employee createEmployee( Employee employee);
	 
	 Employee getEmployeeById( long id);
	 
	 EmployeeDto updateEmployee( long id ,  EmployeeDto employee);
	 
	 EmployeeDto getEmpFromToken(String token);

	
	////////////////////// Tasks ////////////////////
	 
	 List<TaskDto> getTasksByUserId(String token);
	 
	 TaskDto updateTask(long id , TaskDto taskDto );
	 
	 TaskDto getTaskById(long id);

	 TaskDto updateTaskStatus(long id , String status);

	 //////////////////// Comment ///////////////////
	 
	 CommentDto createComment(long taskId , String content,String token);
	 
	 List<CommentDto> getAllComments(long taskId);
	 
	 void deleteComment(long id);
	

}
